nsv13/code/modules/turbolift/lazylift.dm

```diff
@@ -54,8 +55,25 @@ That's it, ok bye!
 	verb_exclaim = "beeps"
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 2
+	var/lift_id = null // A unique ID that can be used to link a specific lift to a specific button
 	var/obj/machinery/lazylift/lift = null //Pointer to the lift controller in our area on this deck, we then ask its master to path the lift to us, assuming it isn't busy or anything.
 
+/obj/machinery/lazylift_button/Initialize()
+	. = ..()
+	if(lift_id)
+		for(var/obj/machinery/lazylift/LL in GLOB.machines)
+			if(LL && (LL.z != z))
+				continue
+			else if(LL.id == lift_id)
+				lift = LL
+	else
+		for(var/turf/T in get_area(src))
+			if(T.z != z)
+				continue
+			var/obj/machinery/lazylift/LL = locate(/obj/machinery/lazylift) in T
+			if(LL)
+				lift = LL
```

Comment:
You can save a bunch of performance by locating the lazylift in the area, rather than each turf

Reply:
Any way to locate multiple items? There should be 2 of these in the area, and I need the one that's on the same z level as the button.

Reply:
There sure is :)
```suggestion
		for(var/obj/machinery/lazylift/LL in get_area(src))
			if(LL.z == z)
				lift = LL
```